### PR TITLE
[v2] Convert `@final` to hard `final`

### PR DESCRIPTION
**Pull request by GromNaN**
Follow-up https://github.com/doctrine/data-fixtures/pull/457

Replace https://github.com/doctrine/data-fixtures/pull/458 (accidentally closed)

### CODE REVIEW

File: phpstan-baseline.neon [hunk 1]
```diff
@@ -65,6 +65,11 @@ parameters:
 			count: 1
 			path: src/Purger/PHPCRPurger.php
 
+		-
+			message: "#^Parameter \\$dm of method Doctrine\\\\Common\\\\DataFixtures\\\\Purger\\\\PHPCRPurgerInterface\\:\\:setDocumentManager\\(\\) has invalid type Doctrine\\\\Common\\\\DataFixtures\\\\Purger\\\\DocumentManager\\.$#"
```
Comment (GromNaN) [inline phpstan-baseline.neon:69]: Skipped due to missing dependency in composer dev-dependencies

**Comment by greg0ire** [comment]
Are there commits that could be reverted independently from the other ones, or should we squash this?

File: src/Purger/PHPCRPurgerInterface.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Common\DataFixtures\Purger;
+
+/**
+ * ORMPurgerInterface
```
Comment (greg0ire) [inline src/Purger/PHPCRPurgerInterface.php]: ?

File: tests/Common/DataFixtures/Executor/ORMExecutorTest.php [hunk 3]
```diff
@@ -135,9 +135,9 @@ private function getMockFixture(): FixtureInterface
         return $this->createMock(FixtureInterface::class);
     }
 
-    /** @return ORMPurger&MockObject */
-    private function getMockPurger(): ORMPurger
+    /** @return ORMPurgerInterface&MockObject */
+    private function getMockPurger(): ORMPurgerInterface
```
Comment (greg0ire) [inline tests/Common/DataFixtures/Executor/ORMExecutorTest.php]: Can't we use a native type declaration here?

File: src/Purger/PHPCRPurgerInterface.php (hunk 2 above)
Comment (GromNaN) [reply src/Purger/PHPCRPurgerInterface.php]: Removed.

File: tests/Common/DataFixtures/Executor/ORMExecutorTest.php (hunk 3 above)
Comment (GromNaN) [reply tests/Common/DataFixtures/Executor/ORMExecutorTest.php]: Yes, updated to use intersection type for every getMock in tests

**Comment by greg0ire** [comment]
It seems there are CI jobs failing. Please take a look at [this guide](https://www.doctrine-project.org/contribute/#dealing-with-checks-and-tools) for more on how to handle those.

**Comment by GromNaN** [comment]
Sorry, I did one wrong change too fast. That should be good.

**Comment by greg0ire** [comment]
It's looking good now. Can you please add something about this in `UPGRADE.md`? :pray:

**Review by greg0ire — APPROVED** [review]

**Review by SenseException — APPROVED** [review]

**Comment by greg0ire** [comment]
Thanks @GromNaN !

**Pull request merged**